### ISSUE TITLE
Getting started with Java APIs

### ISSUE THREAD

**Issue by munenekelvin** [section-engineering-education/engineering-education#3388]
### Proposed title of article
Getting started with Java APIs

### Introduction paragraph (2-3 paragraphs):
API is an abbreviation for Application Programming Interface which is a collection of communication protocols and subroutines used by various programs to communicate between them. A programmer can make use of various API tools to make its program easier and simpler. Also, an API facilitates the programmers with an efficient way to develop their software programs.

Thus in simpler terms, an API helps two programs or applications to communicate with each other by providing them with necessary tools and functions. It takes the request from the user and sends it to the service provider and then again sends the result generated from the service provider to the desired user.

In Java, application programming interfaces (APIs) are predefined software tools that easily enable interactivity between multiple applications. This article offers a detailed explanation of Java APIs.

### Key takeaways: 
By the end of these article the reader will have understood:
- What are Java APIs?
- Who uses Java APIs?
- The need for Java APIs
- Types of Java APIs
- Data and API services
- API service protocols
- List of most commonly used Java APIs
- Advantages of Java APIs

### References: 
N/A

**Comment by WanjaMIKE** [comment]
@munenekelvin 
Good afternoon and thank you for submitting your topic suggestion.
Your topic form has been entered into our queue and should be reviewed (for approval) as soon as a content moderator is finished reviewing the ones in the queue before it.
